Reset board

Opio-Emmanuel-Omona opened the pull request.

Opio-Emmanuel-Omona committed:
add npm and node engines

Opio-Emmanuel-Omona committed:
display board
get the bord from the API

Opio-Emmanuel-Omona committed:
reset board after play
switch turns after play
taost error messages
toast success messages

naush commented on src/actions/responseHandler.js:
```diff
@@ -0,0 +1,33 @@
+    responseHandler.handleResponse(data, symbol, setSymbol)
+  },
+
+  handleResponse(data, symbol, setSymbol) {
```
As discussed in the session, it may be better to encapsulate switching symbols in a `success` callback and error handling in an `error` callback so that response does not need to know the details of how the game works. It would just invoke `success` when `data.message` is Ok, and `error` otherwise.

naush commented on src/actions/restActions.js:
```diff
@@ -0,0 +1,23 @@
+
+const base_url = 'http://localhost:9292/'
+const reset_game_url = base_url+'reset_game';
+const play_url = base_url+'play';
```
Just a reminder from our session, this could be refactored into using environment variables so that you won't have to modify the test when the environment changes.

naush commented on src/components/Button/index.js:
```diff
@@ -0,0 +1,24 @@
+    <div>
+      <button
+        className="btn"
+        style={(props.mark && props.symbol === props.mark.toLowerCase())? btnStyle : []}
```
You can ensure that there is always a default prop value by using the [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html) library.

naush commented on src/components/Button/index.js:
```diff
@@ -0,0 +1,24 @@
+
+function Button(props) {
+  return (
+    <div>
```
You can use `<React.Fragment>` or its shorthand `<>` and `</>` to replace the `div` here unless you have specific styles you want to apply to this tag.

naush commented on src/components/Button/index.test.js:
```diff
@@ -0,0 +1,52 @@
+    wrapper = shallow(<Button />);
+  });
+
+  test('renders without crashing', () => {
```
This duplicates the line from the `beforeEach` block. I feel like it's safe to remove it since it is implied that it will render without crashing with any test this describe block.

naush commented on src/containers/TicTacToe/index.js:
```diff
@@ -0,0 +1,40 @@
+      <div className="control-btns">
+        <Button 
+          mark="X" 
+          handleClick={() => {}} 
```
It might be better to have this as the default behavior for `Button` component. So when you don't pass in an argument for `handleClick`, it either doesn't do anything or trigger an empty body callback.

Opio-Emmanuel-Omona committed:
add ENV variable for base_url
use propTypes to setDefault props
use callbacks to handle success or error on fetch actions
